source/blink.cpp

```diff
@@ -145,9 +148,8 @@ void blink::application::run()
 					}
 				}
 
-				// Fall back to extracting project directory from source files if the link info stream is missing in the PDB
-				if (std::error_code ec; _source_dir.empty() || !std::filesystem::exists(_source_dir, ec))
-					_source_dir = longest_path(cpp_files);
+				// When the project is built out of source, PDB gives weird source directory. Use source common path instead.
```

Comment:
Could you elaborate? The path read from `/LinkInfo` should be the directory the linker was invoked in, which when built via Visual Studio should always be the solution directory.

Reply:
The solution directory is often out of source directory like so (the CMake approach):
```
_build/my_project.sln
sources/foo.cpp
sources/bar.cpp
```
With the current implementation blink is watching `_build` and ignores all the source files.